### PR TITLE
Enhancement for distributed sparse linear regression example 

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
- added the `log_loss` metric used in avazu kaggle competition 
- removed irrelevant code in the LR example
- added a readme file to the sparse folder

@szha

### CODE REVIEW

File: python/mxnet/metric.py [hunk 1]
```diff
@@ -917,6 +917,73 @@ def update(self, labels, preds):
             self.sum_metric += (-numpy.log(prob + self.eps)).sum()
             self.num_inst += label.shape[0]
 
+@register
+@alias('log_loss')
+class LogarithmicLoss(EvalMetric):
```
Comment (piiswrong) [inline python/mxnet/metric.py]: where does the name come from?

File: python/mxnet/metric.py [hunk 2]
```diff
@@ -917,6 +917,73 @@ def update(self, labels, preds):
...
+    """Computes logarithmic loss.
+
+    The cross entropy over a batch of sample size :math:`N` is given by
+
+    .. math::
+       -\\sum_{n=1}^{N}\\sum_{k=1}^{K}t_{nk}\\log (y_{nk}),
+
+    where :math:`t_{nk}=1` if and only if sample :math:`n` belongs to class :math:`k`.
```
Comment (piiswrong) [inline python/mxnet/metric.py]: what's t and y? what's K?

File: python/mxnet/metric.py (hunk 1 above)
Comment (eric-haibin-lin) [reply python/mxnet/metric.py]: The log loss used in avazu kaggle competition 
https://www.kaggle.com/wiki/LogLoss

File: python/mxnet/metric.py (hunk 2 above)
Comment (eric-haibin-lin) [reply python/mxnet/metric.py]: I forgot to update the equation for log loss... Will update it later

**Pull request merged**